File: test_example.js
// Test JavaScript file for narrow-search plugin

function greetUser(name) {
    console.log(`Hello, ${name}!`);
    return `Welcome, ${name}`;
}

const calculateSum = (a, b) => {
    return a + b;
};

class Calculator {
    constructor() {
        this.result = 0;
    }
    
    add(value) {
        this.result += value;
        return this;
    }
    
    multiply(value) {
        this.result *= value;
        return this;
    }
}

function processData(data) {
    // Process some data
    const processed = data.map(item => item * 2);
    return processed;
}

export function exportedFunction() {
    return "I'm exported!";
}

export class ExportedClass {
    method() {
        return "From exported class";
    }
}

File: test_example.py
#!/usr/bin/env python3
"""Test file for the narrow-search plugin."""

def hello_world():
    """Simple function to test narrowing."""
    print("Hello, World!")
    return 42

class TestClass:
    """A test class with methods."""
    
    def __init__(self, name):
        self.name = name
    
    def greet(self):
        """Greet method."""
        print(f"Hello, {self.name}!")
    
    def calculate(self, x, y):
        """Calculate something."""
        result = x + y
        print(f"Result: {result}")
        return result

def another_function(param1, param2):
    """Another function for testing."""
    if param1 > param2:
        return param1
    else:
        return param2

class AnotherClass:
    """Another test class."""
    
    def method_one(self):
        pass
    
    def method_two(self):
        pass
